src/components/dataset/data/DatasetDataView.jsx

```diff
@@ -0,0 +1,220 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import {
+  resetQuery,
+  runQuery,
+  getDatasetById,
+  getDatasetPolicy,
+  countResults,
+  getUserDatasetRoles,
+  pageQuery,
+} from 'actions/index';
+import { FilterList, Info, People } from '@material-ui/icons';
+
+import DataView from 'components/common/data/DataView';
+import LoadingSpinner from 'components/common/LoadingSpinner';
+import DataViewSidebar from './sidebar/DataViewSidebar';
+import InfoView from './sidebar/panels/InfoView';
+import ShareSnapshot from './sidebar/panels/ShareSnapshot';
+import {
+  DATAREPO_ROW_ID_COLUMN_NAME,
+  DATASET_INCLUDE_OPTIONS,
+  GOOGLE_CLOUD_RESOURCE,
+  RESOURCE_TYPE,
+} from '../../../constants';
+
+const QUERY_LIMIT = 1000;
+
+function DatasetDataView({
+  dataset,
+  dispatch,
+  filterStatement,
+  joinStatement,
+  match,
+  orderDirection,
+  orderProperty,
+  profile,
+  queryParams,
+}) {
+  const [selected, setSelected] = useState('');
+  const [selectedTable, setSelectedTable] = useState(null);
+  const [sidebarWidth, setSidebarWidth] = useState(0);
+  const [canLink, setCanLink] = useState(false);
+  const [datasetLoaded, setDatasetLoaded] = useState(false);
+  const [tableNames, setTableNames] = useState([]);
+  const [panels, setPanels] = useState([]);
+
+  useEffect(() => {
+    const datasetId = match.params.uuid;
```

Comment:
We may want to make the datasetId be a property of the component and do the mapping in a `connect` wrapper.

Reply:
I'm not sure how to do that. We are already pulling the match value via the connect wrapper.

There is, however,  a hooks implementation we could use instead: https://v5.reactrouter.com/web/api/Hooks/useparams

Reply:
It also seems like keeping this particular info out of the store makes sense for readability - It's clear that we're triggering updates starting from this component when the url uuid updates.

Reply:
I guess I was thinking of doing this mapping explicitly in the mapStateToProps method?  So at the bottom of the file you might have:

```function mapStateToProps(state) {
  return {
    datasets: state.datasets.datasets,
    datasetsCount: state.datasets.datasetsCount,
    filteredDatasetsCount: state.datasets.filteredDatasetsCount,
    features: state.user.features,
    datasetId: state.params.uuid,
  };
}``

And then datasetId is a first class prop of the component.

Basically, consolidate the mapping there...does that make sense?

Reply:
Yes, that makes sense. 
I'm not sure how to populate the state.params value, though. As far as I can tell, the router state in the store doesn't currently include that information.

Reply:
talked offline and it's totally fine to leave